training/model_management/src/azureml/model/mgmt/processors/transformers/whisper/predict.py

```diff
@@ -160,18 +147,22 @@ def audio_processor(audio_input: str, sampling_rate: int = 16000) -> np.array:
     with TemporaryDirectory() as temp_dir:
         audio_file_path = os.path.join(temp_dir, "audio_file.m4a")
 
-        # check if input string is a URI to an audio file
         if is_valid_url(audio_input):
-            with open(audio_file_path, "wb") as f:
-                f.write(requests.get(audio_input).content)
-            return audio_input_to_nparray(audio_file_path, sampling_rate)
-
-        try:
-            with open(audio_file_path, "wb") as audio:
-                audio.write(base64.b64decode(audio_input))
-        except Exception:
-            raise ValueError("Invalid audio format")
-        return audio_input_to_nparray(audio_file_path, sampling_rate)
+            logging.info("Recieved Remote Audio URL as input.")
+            try:
+                wget.download(audio_input, out=audio_file_path)
+            except Exception:
+                raise ValueError("Invalid URL")
```

Comment:
since this has passed is_valid_url fn, here calling it as invalid URL is confusing ? should this be called as download error